package/gluon-mesh-vpn-tunneldigger/files/lib/gluon/upgrade/400-mesh-vpn-tunneldigger

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/lua
+
+local site = require 'gluon.site_config'
+local users = require 'gluon.users'
+local util = require 'gluon.util'
+
+local uci = require('luci.model.uci').cursor()
+local lutil = require 'luci.util'
+
+-- Group for iptables rule
+users.add_group('gluon-tunneldigger', 900)
+
+local enabled = uci:get_first('tunneldigger', 'broker', 'enabled')
+if (tonumber(enabled) == 0) then
+  enabled = site.tunneldigger_mesh_vpn.enabled and 1 or 0
+end
+
+-- Delete old broker config section       
+uci:delete_all('tunneldigger', 'broker')
+
+section = uci:add('tunneldigger', 'broker')
+
+uci:section('tunneldigger', 'broker', section,
+	  {
+		  enabled = enabled,
+		  uuid = util.node_id(),
+		  interface = 'mesh-vpn',
+		  bind_interface = 'br-wan',
+		  group = 'gluon-tunneldigger',
+		  broker_selection = 'usage',
+		  address = site.tunneldigger_mesh_vpn.brokers,
+	  }
+)
+
+uci:save('tunneldigger')
+uci:commit('tunneldigger')
+
+uci:section('network', 'interface', 'mesh_vpn',
+	  {
+		  ifname = 'mesh-vpn',
+		  proto = 'batadv',
+		  mesh = 'bat0',
+		  mesh_no_rebroadcast = 1,
+		  mtu = site.tunneldigger_mesh_vpn.mtu,
```

Comment:
How is the MAC address assigned for the tunneldigger interface?

As it can't be enabled at the same time as the fastd mesh VPN (as both use the interface name `mesh-vpn`), it could use the same MAC address, I think.